packages/context-editor/src/utils/pubValues.ts

```diff
@@ -1,15 +1,39 @@
 import deepMerge from "deepmerge";
 import { EditorState } from "prosemirror-state";
 
-export const getPubValues = (editorState: EditorState, pubId: string) => {
-	const editedPubs: { [key: string]: { id: string; pubTypeId: string; values: any } } = {};
-	editedPubs[pubId] = {
-		id: pubId,
-		pubTypeId: "67704c04-4f04-46e9-b93e-e3988a992a9b",
+interface DocValue {
+	type: string;
+	attrs: {
+		id: string | null;
+		class: string | null;
+	};
+	content: {
+		type: string;
+		text: string;
+	}[];
+}
+
+interface EditedPubs {
+	[pubId: string]: {
+		parentPubId: string;
+		pubId: string;
+		pubTypeId: string;
 		values: {
-			"rd:content": editorState.doc.toJSON(),
-		},
+			[fieldSlug: string]: DocValue[] | string;
+		};
 	};
+}
+
+export const getPubValues = (editorState: EditorState, pubId: string) => {
```

Comment:
This is the function exported by the ContextEditor to get pub values that should be updated based on the current editor state. You'll notice there's still some hard coded values here, and I commented out the part that sets a "main content" portion to a hard coded slug of `rd:content`. We need to have a wider discussion around what to do here— @tefkah left some good notes about the problem [here](https://docs.google.com/document/d/1y2rsIPXqOXMSYGn4mC42m0qsV1Ag-atkcZeR5WCSuhk/edit?tab=t.0#heading=h.ys4djpwoy1t4). I plan to set up a meeting to discuss this next week!

Reply:
I think this is okay for now! should be revised once we figure out how we want to tackle everything.